libcodechecker/analyze.py

```diff
@@ -0,0 +1,290 @@
+        "checkers - such as the 'core' group - is unsupported by the LLVM/"
+        "Clang community, and thus discouraged.")
+
+    checkers_opts.add_argument('-e', '--enable',
+                               dest="enable",
+                               metavar='checker/checker-group',
+                               default=argparse.SUPPRESS,
+                               action=OrderedCheckersAction,
+                               help="Set a checker (or checker group) "
+                                    "to BE USED in the analysis.")
+
+    checkers_opts.add_argument('-d', '--disable',
+                               dest="disable",
+                               metavar='checker/checker-group',
+                               default=argparse.SUPPRESS,
+                               action=OrderedCheckersAction,
+                               help="Set a checker (or checker group) "
+                                    "to BE PROHIBITED from use in the "
+                                    "analysis.")
+
+    checkers_opts.add_argument('--enable-all',
+                               dest="enable_all",
+                               action='store_true',
+                               required=False,
+                               default=argparse.SUPPRESS,
+                               help="Force the running analyzers to use "
+                                    "almost every checker available. The "
+                                    "checker groups 'alpha.', 'debug.' and "
+                                    "'osx.' (on Linux) are NOT enabled "
+                                    "automatically and must be EXPLICITLY "
+                                    "specified. WARNING! Enabling all "
+                                    "checkers might result in the analysis "
+                                    "losing precision and stability, and "
+                                    "could even result in a total failure of "
+                                    "the analysis. USE WISELY AND AT YOUR "
+                                    "OWN RISK!")
+
+    add_verbose_arguments(parser)
+    parser.set_defaults(func=main)
+
+
+def main(args):
+    """
+    Perform analysis on the given logfiles and store the results in a machine-
+    readable format.
+    """
+
+    context = generic_package_context.get_context()
+
+    # Parse the JSON CCDBs and retrieve the compile commands.
+    actions = []
+
+    for log_file in args.logfile:
+        if not os.path.exists(log_file):
+            LOG.error("The specified logfile '" + log_file + "' does not "
+                      "exist!")
+            continue
+
+        actions += log_parser.parse_log(log_file,
+                                        args.add_compiler_defaults)
+
+    if 'enable_all' in args:
+        # enable_all is handled in
+        # libcodechecker/analyze/analyzers/analyzer_types.py.
```

Comment:
Should we leave this type of comments, what if the filename changes or it will be moved into another module?